### PR TITLE
Add tests for various partition keys

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Add additional apps that have different partition key types. It also adds a barebones admin client that fetches all the imported apps rather than have them passed in as command line arguments.

Fixes https://github.com/realm/realm-dotnet/issues/2093

### CODE REVIEW

File: Tests/Realm.Tests/Sync/SyncConfigurationTests.cs [hunk 1]
```diff
@@ -30,8 +30,7 @@ public class SyncConfigurationTests : SyncTestBase
         [Test]
         public void SyncConfiguration_WithoutPath()
         {
-            var user = GetFakeUser();
```
Comment (nirinchev) [inline Tests/Realm.Tests/Sync/SyncConfigurationTests.cs:33]: The changes here are somewhat unrelated - I restructured the `SyncTestBase` a little and those were caught up in the refactoring.

File: Tests/Realm.Tests/Sync/SyncTestHelpers.cs [hunk 2]
```diff
@@ -132,5 +162,103 @@ public static string[] ExtractBaasSettings(string[] args)
...
+
+            if (appName == "uuid-partition-key")
+            {
+                return AppConfigType.UUIDPartitionKey;
+            }
+
+            throw new NotSupportedException($"Unexpected appName: {appName}");
+        }
+
+        private class BaasClient : IDisposable
```
Comment (nirinchev) [inline Tests/Realm.Tests/Sync/SyncTestHelpers.cs:191]: This is a super barebones helper that calls into the admin API and reads all the apps rather than have them passed in as config parameters. It's intentionally not overengineered/safe because it's not intended to be a general-purpose admin api client.

File: Tests/Realm.Tests/Sync/SynchronizedInstanceTests.cs [hunk 3]
```diff
@@ -89,12 +89,10 @@ public void GetInstanceAsync_ShouldDownloadRealm(bool singleTransaction)
         {
             SyncTestHelpers.RunBaasTestAsync(async () =>
             {
-                var user = await GetUserAsync();
```
Comment (nirinchev) [inline Tests/Realm.Tests/Sync/SynchronizedInstanceTests.cs:92]: This was also caught up in the `SyncTestBase` refactoring.

File: Tests/Realm.Tests/Sync/SyncTestHelpers.cs [hunk 4]
```diff
@@ -75,34 +97,42 @@ public static string[] ExtractBaasSettings(string[] args)
...
                 }
             }
 
-            if (baasUrl != null && baasAppId != null)
+            ExtractBaasApps();
+
+            return result.ToArray();
+        }
+
+        private static void ExtractBaasApps()
```
Comment (nirinchev) [inline Tests/Realm.Tests/Sync/SyncTestHelpers.cs:118]: This gets called either during initial startup (if run via the command line) or the first time a baas test is run. It's synchronous because I didn't want to refactor a whole lot of code around it to make it async and since it's only called once per process, it's not a huge deal.

File: Realm/Realm/Helpers/SerializationHelper.cs [hunk 5]
```diff
@@ -36,6 +36,7 @@ static SerializationHelper()
         {
             var decimalSerializer = new DecimalSerializer(BsonType.Decimal128, new RepresentationConverter(allowOverflow: false, allowTruncation: false));
             BsonSerializer.RegisterSerializer(decimalSerializer);
+            BsonSerializer.RegisterSerializer(new GuidSerializer(GuidRepresentation.Standard));
```
Comment (nirinchev) [inline Realm/Realm/Helpers/SerializationHelper.cs:39]: This serializes uuids as type 4 rather than type 0. Cloud expects 4 and the MongoDB.Bson package defaults to 0 for legacy reasons.

**Comment by papafe** [comment]
It's quite some changes but it seems to make sense
